test/test_sparse_csr.py

```diff
@@ -1080,6 +1086,30 @@ def _test(t, x, y):
                 y = self.genSparseCSRTensor((dk, dj), nnz1, device=device, dtype=dtype, index_dtype=index_dtype)
                 _test(t, x, y)
 
+                # Test mm(CSR, CSR)
+                x = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                y = self.genSparseCSRTensor((dk, di), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                _test_mm(x, y)
+
+                # Test mm(CSR, CSC)
+                x = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                y = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                y = y.transpose(0, 1)
+                _test_mm(x, y)
+
+                # Test mm(CSC, CSR)
+                x = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                y = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                x = x.transpose(0, 1)
+                _test_mm(x, y)
+
+                # Test mm(CSC, CSC)
+                x = self.genSparseCSRTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                y = self.genSparseCSRTensor((dk, di), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
+                x = x.transpose(0, 1)
+                y = y.transpose(0, 1)
```

Comment:
```suggestion
                x = self.genSparseCSCTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
                y = self.genSparseCSCTensor((dk, di), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
```

Reply:
Actually when I try to use this I get
```
  File "/dev/shm/cpuhrsch/pytorch/test/test_sparse_csr.py", line 1120, in test_mm
    test_shape(i, j, k)
  File "/dev/shm/cpuhrsch/pytorch/test/test_sparse_csr.py", line 1105, in test_shape
    x = self.genSparseCSCTensor((di, dk), nnz0, device=device, dtype=dtype, index_dtype=index_dtype)
  File "/dev/shm/cpuhrsch/pytorch/torch/testing/_internal/common_utils.py", line 2076, in genSparseCSCTensor
    return self.genSparseCompressedTensor(size, nnz, layout=torch.sparse_csc, device=device,
  File "/dev/shm/cpuhrsch/pytorch/torch/testing/_internal/common_utils.py", line 2068, in genSparseCompressedTensor
    return torch.sparse_compressed_tensor(compressed_indices, plain_indices,
RuntimeError: (Batch element 0) at position i = 4, the condition ccol_indices[i - 1] <= ccol_indices[i] fails
```
It's not immediately obvious to me what's wrong within `genSparseCompressedTensor`